### ISSUE TITLE
Babel configuration

### ISSUE THREAD

**Issue by robcaldecott** [eirslett/storybook-builder-vite#286]
Is there a more elegant way to configure the `babel` options yet? I am currently hacking my `viteFinal` config to enable the `formatjs` plugin like this:

```js
async viteFinal(config, { configType }) {
    config.plugins = [
      ...config.plugins.filter((plugin) => {
        return !(
          Array.isArray(plugin) && plugin[0].name === "vite:react-babel"
        );
      }),
      require("@vitejs/plugin-react")({
        exclude: [/\.stories\.(t|j)sx?$/, /node_modules/],
        babel: { plugins: ["formatjs"] },
      }),
    ];
    return config;
  },
```

This feels a bit brittle. I'm sure there was talk of a better way to handle this as it seems like a common use case. Anything on the radar?

**Comment by IanVS** [comment]
The storybook options already has a `babel` key.  It seems like we should merge that into the config by default.  I think that would be a better solution, and is closer to the way the webpack builder works, too.

**Comment by mszekiel** [comment]
Can it use `.babelrc` file?

**Comment by mszekiel** [comment]
@IanVS sorry for late reply. For us in the current project, the problem is that vite-builder is using the predefined vitejs config from this repository. And I didn't see any way to provide extended config other than OP posted. 

Maybe additional import of vite config of project inside predefined from builder?

**Comment by IanVS** [comment]
Yes, that's right, the only way right now is with `viteFinal`, and adjusting the plugins manually.  

There are some experiments going on right now that would change the way framework and builder packages work together, and I think if that turns out to work, it'll be a lot easier to set options like this, so I'd like to wait a little longer to see how that shakes out, before we think about adding an option to the global storybook config.  I'd suggest using the approach posted above for now.

**Comment by mszekiel** [comment]
Yes, for now it's a valid solution that works, and we're talking about storybook environment, so it's not affected for production build. 

Looking forward to then, thank you!

**Comment by IanVS** [comment]
For anyone still dealing with this, you can now try out storybook 7 (in alpha currently, not very stable), which uses your `vite.config.js` config as a base for storybook.  So, you can set your babel config in your react plugin there, and storybook will use it.  Try `npx sb@future upgrade --prerelease` to convert.  And let us know if you hit any issues.  There are probably still a few bugs to work out, but it helps solve this particular issue pretty nicely.

**Comment by develohpanda** [comment]
@IanVS that's awesome to hear! Do you know if Storybook 7 has a target stable/release date? Our deployed instance of Storybook is not crucial however I'm wary of jumping onto a prerelease version in case it impacts development cycles.

**Comment by IanVS** [comment]
I haven't heard any recent estimates, but I think it will likely go into beta in October or November, and then stable release at the end of this year or beginning of next.

Totally understandable about not wanting to risk churn with prerelease versions.  I'd love if you were willing to test it out, though, and report any issues you might come across.  That's how we can help ensure the smoothest transition when we release the stable version, through early feedback from users.  :)

**Comment by jcollicoat** [comment]
Hey Ian, we're currently updating our Storybook to v7 alpha with vite, and are experiencing some discrepancies between our `import.meta.env` in when developing vs building. I've created a minimal repo with some examples here: https://github.com/jcollicoat/vite-storybook

I'd appreciate any help I can get on this! Please let me know if I can add more info or if this should be posted elsewhere.

**Comment by IanVS** [comment]
Thanks @jcollicoat, I replied in an issue, https://github.com/jcollicoat/vite-storybook/issues/1, since it's not really related to Babel configuration (this issue here).

**Comment by Psvensso** [comment]
Just to confirm, the vite.config.ts loaded and used by default. 
Work fine to set react > babel config there
Im currently using storybook 7.0.0-beta.19

**Comment by IanVS** [comment]
Yea, in 7.0 it is loaded automatically. This repo is for the 6.5 version, in 7.0 it moved into the main monorepo.

**Comment by karsonkalt** [comment]
Wanting a bit of clarification here, even if my project is on vite, sb7 still needs to use Babel? We are pure esbild and would like to avoid if possible.

**Comment by IanVS** [comment]
SB7 does not use babel in and of itself.  If you're using `@vitejs/plugin-react`, that will use babel.  If you're using `@vitejs/plugin-react-swc` then babel is not involved at all.
